src/transformers/models/vocos/modeling_vocos.py

```diff
@@ -0,0 +1,368 @@
+                        L is the sequence length, and H denotes the model dimension.
+
+        Returns:
+            Tensor: Reconstructed time-domain audio signal of shape (B, T), where T is the length of the output signal.
+            Tensor: Predicted STFT coefficients of shape (B, L, N+2), where N is the number of frequency bins.
+        """
+        x_pred = self.out(x).transpose(1, 2)
+        mag, p = x_pred.chunk(2, dim=1)
+        mag = torch.exp(mag)
+        mag = torch.clip(mag, max=1e2)  # safeguard to prevent excessively large magnitudes
+        # wrapping happens here. These two lines produce real and imaginary value
+        spectrogram_real = torch.cos(p)
+        spectrogram_imag = torch.sin(p)
+        spectrogram_complex = mag * (spectrogram_real + 1j * spectrogram_imag)
+        audio = self.istft(spectrogram_complex)
+        return audio
+
+
+class VocosPreTrainedModel(PreTrainedModel):
+    """
+    An abstract class to handle weights initialization and a simple interface for downloading and loading pretrained
+    models.
+    """
+
+    config_class = VocosConfig
+    base_model_prefix = "vocos"
+    main_input_name = "audio_spectrogram"
+    supports_gradient_checkpointing = False
+
+    def _init_weights(self, module):
+        """Initialize the weights"""
+        if isinstance(module, nn.Linear):
+            std = getattr(self.config, "initializer_range", 0.02)
+            module.weight.data.normal_(mean=0.0, std=std)
+            if module.bias is not None:
+                module.bias.data.zero_()
+        elif isinstance(module, nn.LayerNorm):
+            module.bias.data.zero_()
+            module.weight.data.fill_(1.0)
+        elif isinstance(module, nn.Conv1d):
+            nn.init.kaiming_normal_(module.weight)
+            if module.bias is not None:
+                module.bias.data.zero_()
+        elif isinstance(module, VocosAdaptiveLayerNorm):
+            if hasattr(module, "bias") and module.bias is not None:
+                module.bias.data.zero_()
+            if hasattr(module, "weight") and module.weight is not None:
+                module.weight.data.fill_(1.0)
+
+
+@auto_docstring(
+    custom_intro="""
+    Main Vocos model for neural vocoding for high-quality audio generation. This model can be paired with [`VocosProcessor`] to generate audio from either mel-spectrograms or EnCodec embeddings.
+    """
+)
+class VocosModel(VocosPreTrainedModel):
+    def __init__(self, config: VocosConfig):
+        super().__init__(config)
+        self.backbone = VocosBackbone(config)
+        self.head = VocosISTFTHead(config)
+        self._bandwidth_to_id = {bandwidth: id for id, bandwidth in enumerate(config.bandwidths)}
+        self.post_init()
+
+    @auto_docstring
+    def forward(
+        self,
+        audio_spectrogram: Optional[torch.FloatTensor] = None,
+        input_features: Optional[torch.FloatTensor] = None,
```

Comment:
To two types depending on the model:
- mel-variant: https://huggingface.co/hf-audio/vocos-mel-24khz
- encodec embeddings: https://huggingface.co/hf-audio/vocos-encodec-24khz